### PR TITLE
Add gluecose conformance tests

### PR DESCRIPTION
**Pull request by qmuntal**
This PR checks in several conformance samples from https://github.com/gluecose/test-vectors and test that go-cose can sign and verify them.

Additionally, if the sample allows it, our CBOR output is compared against the sample output.

This new test suite deprecates all tests contained in sign_verify_cose_wg_examples_test.go, which were difficult to run and only did partial validations.

### CODE REVIEW

File: conformance_test.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+	Verify          bool   `json:"shouldVerify"`
+}
+
+type CBOR struct {
+	CBORHex  string `json:"cborHex"`
+	CBORDiag string `json:"cborDiag"`
+}
+
+// Conformance samples are taken from
+// https://github.com/cose-wg/Examples.
```
Comment (thomas-fossati) [inline conformance_test.go:55]: OBE?

**Comment by SteveLasker** [comment]
Awesome seeing these. Did you mean to make two PRs? #19 and #19 ? They seem to impact the same files

**Comment by qmuntal** [comment]
> Awesome seeing these. Did you mean to make two PRs? #19 and #19 ? They seem to impact the same files

This PR supersedes #18. I'll close it when we metge #19 .

**Comment by qmuntal** [comment]
@SteveLasker @thomas-fossati this is ready to merge.

**Review by thomas-fossati — COMMENTED** [review]
Awesome work, 🚢 it!

**Review by thomas-fossati — APPROVED** [review]

**Comment by SteveLasker** [comment]
LGTM

**Pull request merged**